airbyte-server/src/main/java/io/airbyte/server/ServerApp.java

```diff
@@ -256,6 +266,85 @@ public static ServerRunnable getServer(final ServerFactory apiFactory, final Con
     }
   }
 
+  /**
+   * Check that each spec in the database has a spec. If it doesn't, add it. If it can't be added,
+   * track the failure in Segment. The goal is to try to end up in a state where all definitions in
+   * the db contain specs, and to understand what is stopping us from getting there.
+   *
+   * @param configRepository - access to the db
+   * @param schedulerClient - scheduler client so that specs can be fetched as needed
+   */
+  private static void migrateAllDefinitionsToContainSpec(final ConfigRepository configRepository,
```

Comment:
is there a test we can write for this? this is one of those things that if there are edge cases that we hit, it will be really hard to fix (especially after the major version bump).

Reply:
Just added a test suite for this, and I also tested the success and failure flows locally and everything seems to be working as expected. Lmk if you think that is enough or if we need to test more